### PR TITLE
Restrict input types to that get rendered to HTML to not generate HTML errors

### PR DESCRIPTION
**Pull request by sroussey**
Pulled this out and made a draft.

### CODE REVIEW

**Comment by sroussey** [comment]
There are different editors for text, number, color, url, etc.

Also means that default values are checked a bit better than whatever "value" was supposed to mean.

Also, how you name your valueType no longer matters and has no side effect in the HTML

File: lib/components/NodeBaseInputField.tsx [hunk 1]
```diff
@@ -1,101 +1,150 @@
...
+  inputMode,
+  pattern,
+  maxlength,
+  minlength,
+  max,
+  min,
+  step,
+  placeholder,
+  children,
+}: NodeBaseInputFieldProps) => {
```
Comment (clarkmcc) [inline lib/components/NodeBaseInputField.tsx]: Code style: There are enough properties here, and possible more in the future (?) that I would probably just do `...props`, then in the input component do
```jsx
<input {...props}/>
```

File: lib/components/NodeBaseInputField.tsx [hunk 2]
```diff
@@ -1,101 +1,150 @@
...
+  | 'email'
+  | 'month'
+  | 'number'
+  | 'password'
+  | 'range'
+  | 'tel'
+  | 'text'
+  | 'time'
+  | 'url'
+  | 'week'
```
Comment (clarkmcc) [inline lib/components/NodeBaseInputField.tsx]: Can we use `React.HTMLInputTypeAttribute` instead?

File: lib/components/NodeBaseInputField.tsx [hunk 3]
```diff
@@ -1,101 +1,150 @@
...
+
+type NodeBaseInputFieldProps = Pick<NodeInputConfig, 'name'> & {
```
Comment (clarkmcc) [inline lib/components/NodeBaseInputField.tsx]: Can we git rid of a lot of the input props below and do this instead?

```typescript
type NodeBaseInputFieldProps = Pick<NodeInputConfig, 'name'> & React.InputHTMLAttributes<any> & {
...
```

**Review by clarkmcc — CHANGES_REQUESTED** [review]
Huge fan of these changes! Couple code style suggestions

File: lib/components/NodeBaseInputField.tsx (hunk 2 above)
Comment (sroussey) [reply lib/components/NodeBaseInputField.tsx]: Sure. I'll Omit the the one like button, reset, submit, that don't make sense. Or maybe not bother?

Comment (clarkmcc) [reply lib/components/NodeBaseInputField.tsx]: Good catch, sounds like we only support a subset. If that's true then yeah I think an `Omit<...>` makes good sense if you're comfortable with that as well.

**Review by clarkmcc — APPROVED** [review]

**Pull request merged**